vulkan_generator/codegen_utils/codegen.py

```diff
@@ -0,0 +1,154 @@
+
+    # one-line code snippets get commented "in line"
+    if code.count("\n") == 0 or code.count("\n") == 1 and code[-2:-1] == "\n":
+        return code + " /* " +comment + " */\n"
+
+    # multi-line code snippets get a comment above the code
+    return "/* " +comment + " */\n" +code
+
+def create_function_signature(name: str, return_type: str, arguments: Optional[Dict[str, str]]) -> str:
+
+    ret = return_type + " " + name + "("
+
+    for arg_name, arg_type in arguments.items():
+        ret = ret + arg_type + " " + arg_name + ", "
+
+    # Strip the trailing comma from the last argument
+    if arguments:
+        ret = ret[0:-2]
+
+    ret = ret + ")"
+
+    return ret
+
+def create_function_declaration(name: str,
+                                return_type: str = "void",
+                                arguments: Optional[Dict[str, str]] = None) -> str:
+    return create_function_signature(name, return_type, arguments,) + ";"
+
+def create_function_definition(name: str,
+                               return_type: str = "void",
+                               arguments: Optional[Dict[str, str]] = None,
+                               code: str = "") -> str:
+    return create_function_signature(name, return_type, arguments) + " {\n" + indent_code(code) +"\n}"
+
+def create_exception_declaration(name: str, base_class: str = "std::exception") -> str:
+    return "class " + name + " : public " + base_class +" {};"
+
+def create_class_definition(name: str,
+                            public_inheritance: Optional[List[str]] = None,
+                            protected_inheritance: Optional[List[str]] = None,
+                            private_inheritance: Optional[List[str]] = None,
+                            public_members: Optional[List[str]] = None,
+                            protected_members: Optional[List[str]] = None,
+                            private_members: Optional[List[str]] = None,
+                            public_functions: Optional[List[str]] = None,
+                            protected_functions: Optional[List[str]] = None,
+                            private_functions: Optional[List[str]] = None
+                            ) -> str:
+
+    def inheritance_helper(inheritance_type : str, inheritance : List[str]) -> str:
+        if not inheritance:
+            return ""
+        inheritance_string = inheritance_type + " "
+        for base in inheritance:
+            inheritance_string = inheritance_string + base + ", "
+        return inheritance_string
+
+    def member_helper(member_type : str, members : List[str]) -> str:
+        if not members:
+            return ""
+        members_string = indent() + member_type +":\n"
+        for member in members:
+            members_string = members_string + indent(2) + member + "\n"
+        return members_string + "\n"
+
+    ret = "class " + name
+
+    inheritance = public_inheritance or protected_inheritance or private_inheritance
```

Comment:
maybe we may separate these to functions.

Reply:
As discussed we can look at this but the code will stand for now